lib/rspec-console/config_cache.rb

```diff
@@ -1,82 +1,84 @@
+# We have to reset the RSpec.configuration, because it contains a lot of
+# information related to the current test (what's running, what are the
+# different test results, etc).
+#
+# RSpec.configuration gets also loaded with a bunch of stuff from the
+# 'spec/spec_helper.rb' file. Often that instance is extended with other
+# modules (FactoryGirl, Mocha,...) and we don't want to replace requires with
+# load all around the place.
+#
+# Instead, we cache whatever is done to RSpec.configuration during the
+# first invokration of require('spec_helper').
+# This is done by interposing the Proxy class on top of RSpec.configuration.
 class RSpecConsole::ConfigCache
-  # We have to reset the RSpec.configuration, because it contains a lot of
-  # information related to the current test (what's running, what are the
-  # different test results, etc).
-  #
-  # RSpec.configuration gets also loaded with a bunch of stuff from the
-  # 'spec/spec_helper.rb' file. Often that instance is extended with other
-  # modules (FactoryGirl, Mocha,...) and we don't want to replace requires with
-  # load all around the place.
-  #
-  # Instead, we cache whatever is done to RSpec.configuration during the
-  # first invokration of require('spec_helper').
-  # This is done by interposing the Proxy class on top of RSpec.configuration.
-  #
-  attr_accessor :proxy, :recorded_config, :shared_examples_groups
-
-  def initialize
-    ::RSpec.instance_eval do
-      def self.configuration=(value)
-        @configuration = value
-      end
-    end
-  end
+  attr_accessor :proxy, :recorded_registry
 
   def cache
-    if self.proxy
-      # replay
-      ::RSpec.configure do |config|
-        self.recorded_config.each do |msg|
-          config.send(msg[:method], *msg[:args], &msg[:block])
+    if have_recording?
+      ::RSpec.configure &replay_recorded_config
+
+      # RSpec 2 and 3 have different APIs for accessing shared_examples. 3 has
+      # the concept of a "registry" whereas 2 does not.
+      if version >= Gem::Version.new('3')
+        unless shared_examples.empty?
+          ::RSpec.world.
+            shared_example_group_registry.
+          add(
+            :main,
+            shared_examples.keys.first,
+            &shared_examples.values.first
+          )
```

Comment:
i disagree on the readability, imo, this is more readable than parsing a gigantic chain of functions.